internal/armada/repository/job.go

```diff
@@ -838,36 +839,60 @@ func (repo *RedisJobRepository) GetQueueJobIds(queueName string) ([]string, erro
 }
 
 func (repo *RedisJobRepository) GetActiveJobIds(queue string, jobSetId string) ([]string, error) {
+	return repo.GetJobSetJobIds(queue, jobSetId, &JobSetFilter{
+		IncludeLeased: true,
+		IncludeQueued: true,
+	})
+}
+
+type JobSetFilter struct {
+	IncludeQueued bool
+	IncludeLeased bool
+}
+
+func (repo *RedisJobRepository) GetJobSetJobIds(queue string, jobSetId string, filter *JobSetFilter) ([]string, error) {
+	var queuedIdsCommandResult *redis.StringSliceCmd
+	var leasedIdsCommandResult *redis.StringSliceCmd
 
 	tx := repo.db.TxPipeline()
-	queuedIdsCommand := tx.ZRange(jobQueuePrefix+queue, 0, -1)
-	leasedIdsCommand := tx.ZRange(jobLeasedPrefix+queue, 0, -1)
+	if filter == nil || filter.IncludeQueued {
+		queuedIdsCommandResult = tx.ZRange(jobQueuePrefix+queue, 0, -1)
+	}
+	if filter == nil || filter.IncludeLeased {
+		leasedIdsCommandResult = tx.ZRange(jobLeasedPrefix+queue, 0, -1)
+	}
 	jobSetIdsCommand := tx.SMembers(jobSetPrefix + jobSetId)
 
 	_, err := tx.Exec()
 	if err != nil {
 		return nil, errors.WithStack(err)
 	}
 
-	queuedIds, err := queuedIdsCommand.Result()
-	if err != nil {
-		return nil, errors.WithStack(err)
+	activeJobIds := []string{}
+	if filter == nil || filter.IncludeQueued {
+		queuedIds, err := queuedIdsCommandResult.Result()
+		if err != nil {
+			return nil, errors.WithStack(err)
+		}
+		activeJobIds = append(activeJobIds, queuedIds...)
 	}
-
-	leasedIds, err := leasedIdsCommand.Result()
-	if err != nil {
-		return nil, errors.WithStack(err)
+	if filter == nil || filter.IncludeLeased {
+		leasedIds, err := leasedIdsCommandResult.Result()
+		if err != nil {
+			return nil, errors.WithStack(err)
+		}
+		activeJobIds = append(activeJobIds, leasedIds...)
 	}
 
 	jobSetIds, err := jobSetIdsCommand.Result()
 	if err != nil {
 		return nil, errors.WithStack(err)
 	}
 
-	activeIds := util.StringListToSet(append(queuedIds, leasedIds...))
+	activeIdsSet := util.StringListToSet(activeJobIds)
 	activeSetIds := []string{}
 	for _, id := range jobSetIds {
-		if activeIds[id] {
+		if activeIdsSet[id] {
```

Comment:
the naming here is a bit confusing. We have `activeIdsSet` (which is a set) and `activeSetIds` (which is logically a set).